hs/src/Reach/AST/SL.hs

```diff
@@ -160,6 +160,38 @@ data SLVal
   | SLV_Deprecated Deprecation SLVal
   deriving (Eq, Generic)
 
+-- | Equivalence operation on flattened or simplified structures.
+class Equiv a where
+  equiv :: a -> a -> Bool
```

Comment:
there's probably no reason to typeclass this atm but I figured semantically there is some use for a shallow equivalence class? Otherwise this would be fine as a simple function

Reply:
Yea, I think it should be a class